### PR TITLE
UI: Driver health checking

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
This adds a few things:

1. Client nodes with unhealthy drivers get an icon on the clients list page
2. Client nodes show the number of unhealthy drivers in the attributes strip on the detail page
3. Allocations on the client detail page get an icon when they depend on a driver that is unhealthy
4. The client detail page has a new node events section to list the last ten events for the node
5. The client detail page has a new drivers section that shows all drivers, if they are detected on the node, if they are healthy on the node, and when they were last updated.
6. The drivers on the client detail page can be expanded to show the health description as well as the driver attributes (e.g., version).
7. Allocations on the task group page get an icon as well when they depend on an unhealthy driver
8. Tasks on the allocation detail page get an icon when they depends on an unhealthy driver.

screen shot 2018-05-21 at 11 02 46 am

_Allocations on the task group and client pages are denoted when they depend on an unhealthy driver_

screen shot 2018-05-21 at 11 05 12 am

_The client detail page now shows all drivers and their status, (i.e., detected and healthy)_

screen shot 2018-05-21 at 11 13 47 am

_Drivers can be expanded to show details including driver-specific attributes and a message regarding health_

screen shot 2018-05-21 at 11 14 14 am

_Client events include the subsystem the event is for and a message_

### CODE REVIEW

File: ui/app/components/list-accordion.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import Component from '@ember/component';
+import { computed, get, set } from '@ember/object';
+
+export default Component.extend({
+  classNames: ['accordion'],
+
+  key: 'id',
+  source: computed(() => []),
```
Comment (johncowen) [inline ui/app/components/list-accordion.js]: Ah not seen this, I've been doing it in `init` I think, nice!

File: ui/app/mixins/with-component-visibility-detection.js [hunk 2]
```diff
@@ -7,11 +8,15 @@ export default Mixin.create({
   },
 
   setupDocumentVisibility: function() {
-    this.set('_visibilityHandler', this.get('visibilityHandler').bind(this));
-    document.addEventListener('visibilitychange', this.get('_visibilityHandler'));
+    if (!Ember.testing) {
+      this.set('_visibilityHandler', this.get('visibilityHandler').bind(this));
+      document.addEventListener('visibilitychange', this.get('_visibilityHandler'));
+    }
```
Comment (johncowen) [inline ui/app/mixins/with-component-visibility-detection.js]: Is this similar to that on click outside not cleaning up thing I had during testing?

File: ui/app/templates/components/list-accordion.hbs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{{#each decoratedSource as |item|}}
+  {{yield (hash
+    head=(component "list-accordion/accordion-head"
+      isOpen=item.isOpen
+      onOpen=(action "open" item)
+      onClose=(action "close" item))
```
Comment (alisdair) [inline ui/app/templates/components/list-accordion.hbs]: Could use `(action (mut item.isOpen) true)` and `(action (mut item.isOpen) false)` here instead, I think.

File: ui/app/models/job.js [hunk 4]
```diff
@@ -116,6 +116,28 @@ export default Model.extend({
   evaluations: hasMany('evaluations'),
   namespace: belongsTo('namespace'),
 
+  drivers: computed('taskGroups.@each.drivers', function() {
+    return this.get('taskGroups')
+      .mapBy('drivers')
+      .reduce((all, drivers) => {
+        all.push(...drivers);
+        return all;
+      }, [])
```
Comment (alisdair) [inline ui/app/models/job.js]: If only we had `Array.prototype.smoosh`…

File: ui/app/serializers/node.js [hunk 5]
```diff
@@ -9,11 +11,10 @@ export default ApplicationSerializer.extend({
   },
 
   normalize(modelClass, hash) {
-    // Proxy local agent to the same proxy express server Ember is using
-    // to avoid CORS
-    if (this.get('config.isDev') && hash.HTTPAddr === '127.0.0.1:4646') {
-      hash.HTTPAddr = '127.0.0.1:4200';
-    }
+    // Transform the map-based Drivers object into an array-based ClientDriver fragment list
```
Comment (alisdair) [inline ui/app/serializers/node.js]: This comment refers to `ClientDriver`, but I think the fragment model name is `NodeDriver`. Am I mistaken?

File: ui/app/mixins/with-component-visibility-detection.js (hunk 2 above)
Comment (DingoEatingFuzz) [reply ui/app/mixins/with-component-visibility-detection.js]: This had to do with requests canceling and resulting in runloop errors being thrown in tests when I was multitasking.

File: ui/app/templates/components/list-accordion.hbs (hunk 3 above)
Comment (DingoEatingFuzz) [reply ui/app/templates/components/list-accordion.hbs]: Good call. I always forget about `mut`.

File: ui/app/serializers/node.js (hunk 5 above)
Comment (DingoEatingFuzz) [reply ui/app/serializers/node.js]: Ugh, yeah. We use Node and Client sorta but not really interchangeably. I try to keep all the data layers stuff using Node and all the routes and copy saying Client. Really easy to get wires crossed.

**Pull request merged**